api/task_status.go

```diff
@@ -0,0 +1,155 @@
+package api
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/hashicorp/consul-terraform-sync/event"
+)
+
+const taskStatusPath = "status/tasks"
+
+// TaskStatus is the status for a single task
+type TaskStatus struct {
+	TaskName  string   `json:"task_name"`
+	Status    string   `json:"status"`
+	Providers []string `json:"providers"`
+	Services  []string `json:"services"`
+	EventsURL string   `json:"events_url"`
+}
+
+// taskStatusHandler handles the task status endpoint
+type taskStatusHandler struct {
+	store   *event.Store
+	version string
+}
+
+// newTaskStatusHandler returns a new TaskStatusHandler
+func newTaskStatusHandler(store *event.Store, version string) *taskStatusHandler {
+	return &taskStatusHandler{
+		store:   store,
+		version: version,
+	}
+}
+
+// ServeHTTP serves the task status endpoint which returns a map of taskname to
+// task status
+func (h *taskStatusHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	log.Printf("[TRACE] (api.taskstatus) requesting task status '%s'", r.URL.Path)
+
+	taskName, err := getTaskName(r.URL.Path, h.version)
+	if err != nil {
+		log.Printf("[TRACE] (api.taskstatus) bad request: '%s'", err)
```

Comment:
nit to remove the single quotes around the error string, I think the colon before it makes it clear it's a new component to the log

Reply:
👍